### PR TITLE
Added detailed error handler functions for CRUD operations on resources and data sources

### LINKED ISSUE #28: Modifying SDDC name in GUI doesn't reflect on Terraform
After changing the name of my Terraform deployed SDDC from the GUI and later running `Terraform Refresh` to update the configuration, there is no update happening and the .tfstate file continues to show original SDDC name.

### PR DESCRIPTION
**Pull request by saloneerege**
@rchadalawada @shylasrinivas

### CODE REVIEW

**Review by rchadalawada — COMMENTED** [review]
LGTM

File: vmc/error_handler.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	return logAPIError(msg, err)
+}
+
+func HandleUpdateError(resourceType string, err error) error {
+	msg := fmt.Sprintf("Failed to update %s", resourceType)
+	return logAPIError(msg, err)
+}
+
+func HandleListError(resourceType string, err error) error {
+	msg := fmt.Sprintf("Failed to read %ss", resourceType)
```
Comment (shylasrinivas) [inline vmc/error_handler.go]: %ss?

Comment (saloneerege) [reply vmc/error_handler.go]: Good catch. Will make the change

**Review by shylasrinivas — APPROVED** [review]

File: vmc/error_handler.go (hunk 1 above)
Comment (saloneerege) [reply vmc/error_handler.go]: Done

**Pull request merged**